src/iceberg_functions/iceberg_multi_file_reader.cpp

```diff
@@ -399,17 +440,48 @@ void IcebergMultiFileReader::FinalizeChunk(ClientContext &context, const MultiFi
                                            DataChunk &input_chunk, DataChunk &output_chunk,
                                            ExpressionExecutor &executor,
                                            optional_ptr<MultiFileReaderGlobalState> global_state) {
-	// Base class finalization first
+
+	//! Add the extra equality delete fields to output chunk.
+	int32_t diff = 0;
+	if (executor.expressions.size() != output_chunk.ColumnCount()) {
+		diff = executor.expressions.size() - output_chunk.ColumnCount();
+		for (int32_t i = diff; i > 0; i--) {
+			int32_t index = input_chunk.ColumnCount() - i;
+			output_chunk.data.emplace_back(input_chunk.data[index]);
+		}
+	}
+
+	//! Base class finalization first
 	MultiFileReader::FinalizeChunk(context, bind_data, reader, reader_data, input_chunk, output_chunk, executor,
 	                               global_state);
 
+	//! Map from index into local_columns -> field_id
+	auto &local_columns = reader.columns;
+	unordered_map<idx_t, idx_t> column_index_to_field_id;
+	for (idx_t i = 0; i < local_columns.size(); i++) {
+		auto &col = local_columns[i];
+		column_index_to_field_id[i] = col.identifier.GetValue<int32_t>();
+	}
+
+	//! Map from field_id -> index in 'output_chunk'
+	unordered_map<idx_t, idx_t> field_id_to_result_id;
+	auto &column_indexes = reader.column_indexes;
+	auto result_id = executor.expressions.size() - column_indexes.size();
+	for (auto &column_index : column_indexes) {
+		field_id_to_result_id[column_index_to_field_id[column_index.GetPrimaryIndex()]] = result_id++;
```

Comment:
I reworked this logic because I couldn't find a good reason for iterating backwards, and the code was quite dense because of it
Can you double-check that this behaves the same as your original logic?